Add globalization API breaking change (#18492)
dotnet/docs: docs/core/run-time-config/globalization.md

---
title: Globalization config settings

> description: Learn about run-time settings that configure globalization aspects of a .NET Core app, for example, how it parses Japanese dates.

Reviewer: ```suggestion
description: Learn about runtime settings that configure globalization aspects of a .NET Core app, for example, how it parses Japanese dates.
```

ms.date: 05/18/2020
ms.topic: reference
---

> # Run-time configuration options for globalization

Reviewer: ```suggestion
# Runtime configuration options for globalization
```

| - | - | - |
| **runtimeconfig.json** | `Switch.System.Globalization.FormatJapaneseFirstYearAsANumber` | `false` - format as "Gannen"<br/>`true` - format as number |
| **Environment variable** | N/A | N/A |

## NLS

> - Determines whether .NET uses National Language Support (NLS) or International Components for Unicode (ICU) globalization APIs for Windows apps. .NET 5.0 and later versions use ICU globalization APIs by default on Windows 10.

Reviewer 2: Might be worth mentioning that this is on Windows 10 19H1 or later?

> - Default: Use ICU globalization APIs (`false`).

Reviewer 2: This should instead be: 
```suggestion
- Default: Use NLS globalization APIs (`false`).
```

or 

```suggestion
- Default: Use ICU globalization APIs (`true`).
```

Author: This is trying to show the default value if you don't explicitly set it, which is to use ICU, correct?

Reviewer 2: Right, if the switch is not present, the default behavior is to use ICU.

Author: So should I leave the default out altogether? Your suggestion is not making sense to me.

Reviewer 2: Now I understand what this is saying. However the (`false`) inside the parenthesis was confusing.

but this is saying that the default is to use ICU... maybe remove the `false` part?

Author: Thanks @safern. I agree that it might be confusing as written. Just to understand better, what did you think the `false` referred to? I'm trying to devise a better way to display defaults for this setting and all the others that currently list it in the same way.

Reviewer 2: I understood `false` referred to `Use ICU globalization APIs` — but after your response I was then able to understand it referred to using NLS and that the default was referring to it using ICU. 

Does that make sense?

Author: Yes, that makes sense, thanks!

- For more information, see [Globalization APIs use ICU libraries on Windows](../compatibility/3.1-5.0.md#globalization-apis-use-icu-libraries-on-windows).

> | | Setting name | Values | Introduced |

Reviewer 2: It seems like this table has a syntax error:

![image](https://user-images.githubusercontent.com/22899328/82269313-2d894800-9926-11ea-8648-25603fe1bdef.png)

Author: Oops, thanks.
